src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLogger.cs

```diff
@@ -308,39 +343,45 @@ private void LogFunctionResultAggregate(IEnumerable<KeyValuePair<string, object>
                         // We won't use the format string here
                         break;
                     default:
-                        if (value.Value is TimeSpan)
+                        if (value.Value is int intValue) 
                         {
-                            // if it's a TimeSpan, log the milliseconds
-                            metrics.Add(value.Key, ((TimeSpan)value.Value).TotalMilliseconds);
+                            metrics.Add(value.Key, Convert.ToDouble(intValue));
                         }
-                        else if (value.Value is double || value.Value is int)
+                        else if (value.Value is double doubleValue)
                         {
-                            metrics.Add(value.Key, Convert.ToDouble(value.Value));
+                            metrics.Add(value.Key, doubleValue);
+                        }
+                        else if (value.Value is TimeSpan timeSpan)
+                        {
+                            // if it's a TimeSpan, log the milliseconds
+                            metrics.Add(value.Key, timeSpan.TotalMilliseconds);
                         }
-
                         // do nothing otherwise
                         break;
                 }
             }
 
+            IDictionary<string, string> properties = new Dictionary<string, string>(2);
+            ApplyKnownProperties(properties, logLevel, eventId);
+
             foreach (KeyValuePair<string, double> metric in metrics)
             {
-                _telemetryClient.TrackMetric($"{functionName} {metric.Key}", metric.Value);
+                _telemetryClient.TrackMetric($"{functionName} {metric.Key}", metric.Value, properties);
```

Comment:
Is `metric.Key` a finite set of values? Allocations per-concat here would be something we can eliminate if so

Reply:
yeah, but functionName varies as well.